packages/babel-plugin-transform-spread/src/index.ts

```diff
@@ -175,7 +175,12 @@ export default declare((api, options: Options) => {
         if (t.isMemberExpression(callee)) {
           const temp = scope.maybeGenerateMemoised(callee.object);
           if (temp) {
-            callee.object = t.assignmentExpression("=", temp, callee.object);
+            callee.object = t.assignmentExpression(
+              "=",
+              temp,
+              // object must not be Super when `temp` is an identifier
+              callee.object as t.Expression,
```

Comment:
It is pretty annoying that we can't have TS narrowed down the `object` type in such pattern `const foo = scope.maybeGenerateMemoised(object); if (foo) { // object must not be Super | TopicReference ... }`. I tried conditional type like
```ts
maybeGenerateMemoised<T extends t.Node>(object: T): T extends t.Super ? null : t.Identifier
```

TS can infer `maybeGenerateMemoised` returns `null` when `object` is `Super`, but it can't infer `T` from the return results.